src/lexer.jl

```diff
@@ -215,15 +216,23 @@ end
 Returns a `Token` of kind `kind` with contents `str` and starts a new `Token`.
 """
 function emit(l::Lexer, kind::Kind, err::TokenError = Tokens.NO_ERR)
-    if kind == Tokens.IDENTIFIER || isliteral(kind) || kind == Tokens.ERROR
+    if !l.raw && (kind == Tokens.IDENTIFIER || isliteral(kind) || kind == Tokens.COMMENT || kind == Tokens.WHITESPACE)
         str = String(take!(l.charstore))
+    elseif kind == Tokens.ERROR
+        str = String(l.io.data[(l.token_startpos + 1):position(l.io)])
     else
         str = ""
     end
-    tok = Token(kind, (l.token_start_row, l.token_start_col),
-                (l.current_row, l.current_col - 1),
-                startpos(l), position(l) - 1,
-                str, err)
+    if l.raw
```

Comment:
This will not be type stable right.

Reply:
Yep return type is a Union, I can just parameterize `Lexer` if you like